onmt/tests/test_beam.py

```diff
@@ -0,0 +1,371 @@
+import unittest
+from onmt.translate.beam import Beam
+
+import torch
+
+
+class GlobalScorerStub(object):
+    def update_global_state(self, beam):
+        pass
+
+    def score(self, beam, scores):
+        return scores
+
+
+class TestBeam(unittest.TestCase):
+    BLOCKED_SCORE = -10e20
+
```

Comment:
here -10e20
later -1e20
do we need to switch to stuff like -inf for future fp16 support or it's a none issue ?

Reply:
The scores should always be in FP32.

Reply:
ok. but we have a bit of everything in the code like here https://github.com/OpenNMT/OpenNMT-py/blob/master/onmt/modules/multi_headed_attn.py#L172
just a comment, if you think it's ok, no big deal.